EcpBtBot/MainClass.py

```diff
@@ -1,39 +1,42 @@
 
 import os
+import logging
 from .DialogFlow.API import *
 from .DialogFlow.DialogFlowResponse import *
 from .Conversational_Integration import *
 
-print("Dire Bonjour pour commencer \n")
 
 dico_users = {}
 #dictionnaire (var globale): pour chaque clé (user), on a un dictionnaire de connexions API en fonction des thèmes
 
 class API_reponse:
 
     def __init__(self, sujet):
-        if sujet == "sepresenter":
+        if sujet == "Se_presenter":
```

Comment:
you can also have a dictionary, which keys are the subjects and values are the different object API_x(). This dictionary can be inside the class but outside the init (since it's the same for any instance). Then inside the init, you just check if the subject is inside this dictionary, and if it is, self.ai is the corresponding value. 

Actually this class seems to be following the Factory pattern (it's a standard concept of design pattern in object oriented programming)

Reply:
ok, we are refactoring the creation of API that will be linked to the subject via a dictionary.
The Client Token will be an entry for the creation of the APIs. We need to create a global dico with keys=subject and value= client_token